### PR TITLE
Fix an issue of GFX11 GS transform feedback 

### PR DESCRIPTION
**Pull request by amdrexu**
There are tests that do rasterization yet still do transform feedback.
The numbers of primitives generated are queried both for raster stream
and for non-raster stream. The GLSL source is something like this:

```
  layout(points) in;
  layout(points, max_vertices = 2) out;
  layout(xfb_buffer = 0, xfb_offset = 0, xfb_stride = 16,
         location = 0, stream = 1) out vec4 xfb;
  void main (void) {
    gl_PointSize = 1.0;
    xfb = vec4(42);
    EmitStreamVertex(0);
    EndStreamPrimitive(0);
    EmitStreamVertex(1);
    EndStreamPrimitive(1);
  }
```

There are two streams, one is for rasterization but is not associated
with XFB and the other is dedicated for XFB. The number of generated
primitives are finally queried. Therefore, we must treat vertex streams
as active if they are associated with XFB buffer or is raster stream
itself. Otherwise, the raster stream is not processed.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit a5c9158f4ee48f4a56cfa48c9a09e34a2e761384
<details><summary>CTS tests (Failed: 0/211498)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39250/70506 (55.7%)</li>
<li>Failed: 0/70506 (0.0%)</li>
<li>Not Supported: 31256/70506 (44.3%)</li><li>Warnings: 0/70506 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70495 (56.0%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 30997/70495 (44.0%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36345/70497 (51.6%)</li>
<li>Failed: 0/70497 (0.0%)</li>
<li>Not Supported: 34152/70497 (48.4%)</li><li>Warnings: 0/70497 (0.0%)</li>
</ul></details>


</ul></details>

**Review by ruiling — COMMENTED** [review]
Can we possibly add a test for this?

**Comment by amdrexu** [comment]
> Can we possibly add a test for this?

Yes, I plan to do it in another PR because we have several GFX11 tests in the internal repo that need to be ported to GPUOpen.

File: lgc/patch/NggPrimShader.cpp [hunk 1]
```diff
@@ -2179,7 +2179,8 @@ void NggPrimShader::buildPrimShaderWithGs(Function *entryPoint) {
     if (m_pipelineState->enableSwXfb()) {
       const auto &streamXfbBuffers = m_pipelineState->getStreamXfbBuffers();
       for (unsigned i = 0; i < MaxGsStreams; ++i) {
-        bool streamActive = streamXfbBuffers[i] != 0;
+        // Treat the vertex stream as active if it is associated with XFB buffers or is the rasterization stream.
```
Comment (linqun) [inline lgc/patch/NggPrimShader.cpp:2182]: the comment doesn't match with the code change. could you remove the comment or refine it? thanks!

Comment (amdrexu) [reply lgc/patch/NggPrimShader.cpp:2182]: Oh, my carelessness.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5de2fba8c69cac16e90f92f411bcba487bc2b866
<details><summary>CTS tests (Failed: 0/211487)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39248/70495 (55.7%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 31247/70495 (44.3%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39498/70495 (56.0%)</li>
<li>Failed: 0/70495 (0.0%)</li>
<li>Not Supported: 30997/70495 (44.0%)</li><li>Warnings: 0/70495 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36345/70497 (51.6%)</li>
<li>Failed: 0/70497 (0.0%)</li>
<li>Not Supported: 34152/70497 (48.4%)</li><li>Warnings: 0/70497 (0.0%)</li>
</ul></details>


</ul></details>

**Review by linqun — APPROVED** [review]

**Pull request merged**